File: CPP [1501-2000]/1561.maximum_number_of_coins_you_can_get [1].cpp
class Solution {
public:
    int maxCoins(vector<int>& piles) { // T.C.=O(n*logn), S.C.=O(1)
        sort(piles.begin(), piles.end());
        int n=piles.size() / 3;
        int sum = 0;
        for(int i=3*n-2; i >= n; i -= 2)
            sum += piles[i];
        return sum;
    }
};

File: Javascript/1561.maximum-number-of-coins-you-can-get [1].js
/**
 * @param {number[]} piles
 * @return {number}
 */
var maxCoins = function(piles) {
    piles.sort((a, b) => a-b)

    let sum = 0
    for(let i=0, j=piles.length-1; i<j; i++, j-=2)
        sum += piles[j-1]

    return sum
};